client/tests_start_https.go

```diff
@@ -0,0 +1,565 @@
+XNZg0PjPJ5X1i7+Jjb5d9sMMwIMhAr2WYEnzxZ3FZaHSEneV0j3qNOjEAzoTAPRd
+/Unrk3uuY1oor5f0EIHPddcATpfaNCaQt58zxtoVA8f/y1r9WXyvnGMT58MioZdC
+ietH+CUhJnb2aHkpou8WeYP91OTi9C7zlKWTNyjkv7vE0ocnghrzsQTtVZ/AiJhi
+VATCc/b7dmSHzML+waYlIMXue57JocQMCvIv/ho5e5MTA6CZYQ/QSQC2Q/GjYt0i
+7ObbxfRZ5Ohdyb0X9FncuD6uty/lJXqfaKlQAP5OnvXOqEYROayhEMfY7924M44E
+vJfJnpI6fBABEGNuw1thZaLqY7/ce7dHDBwgyNtIKnnj/I7YesTagFesR+JoQJGW
+V7XUCP//RRP4QKG0BNdCL3iUFR72MebQy3wV29VFI7y32wv8H9wKHg5qgoWfdQyE
+yeFwqtSOng0fHURzLeB0PjuU/hjjEHzfIzf5W4zoXdY3ngxX1SYXIzD36QVxnsiG
+w1bAMxN9o2V8Be1gZCw1S1MtVK1+lvgsS2rKrWz0xfpYriWarWfsmw==
+-----END RSA PRIVATE KEY-----`)
+
+var localhostCertShortEEKey = []byte(`-----BEGIN CERTIFICATE-----
+MIICEzCCAXygAwIBAgIQMIMChMLGrR+QvmQvpwAU6zANBgkqhkiG9w0BAQsFADAS
+MRAwDgYDVQQKEwdBY21lIENvMCAXDTcwMDEwMTAwMDAwMFoYDzIwODQwMTI5MTYw
+MDAwWjASMRAwDgYDVQQKEwdBY21lIENvMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCB
+iQKBgQDuLnQAI3mDgey3VBzWnB2L39JUU4txjeVE6myuDqkM/uGlfjb9SjY1bIw4
+iA5sBBZzHi3z0h1YV8QPuxEbi4nW91IJm2gsvvZhIrCHS3l6afab4pZBl2+XsDul
+rKBxKKtD1rGxlG4LjncdabFn9gvLZad2bSysqz/qTAUStTvqJQIDAQABo2gwZjAO
+BgNVHQ8BAf8EBAMCAqQwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDwYDVR0TAQH/BAUw
+AwEB/zAuBgNVHREEJzAlggtleGFtcGxlLmNvbYcEfwAAAYcQAAAAAAAAAAAAAAAA
+AAAAATANBgkqhkiG9w0BAQsFAAOBgQCEcetwO59EWk7WiJsG4x8SY+UIAA+flUI9
+tyC4lNhbcF2Idq9greZwbYCqTTTr2XiRNSMLCOjKyI7ukPoPjo16ocHj+P3vZGfs
+h1fIw3cSS2OolhloGw/XM6RWPWtPAlGykKLciQrBru5NAPvCMsb/I1DAceTiotQM
+fblo6RBxUQ==
+-----END CERTIFICATE-----`)
+
+var localhostKeyShortEEKey = []byte(`-----BEGIN RSA PRIVATE KEY-----
+MIICXgIBAAKBgQDuLnQAI3mDgey3VBzWnB2L39JUU4txjeVE6myuDqkM/uGlfjb9
+SjY1bIw4iA5sBBZzHi3z0h1YV8QPuxEbi4nW91IJm2gsvvZhIrCHS3l6afab4pZB
+l2+XsDulrKBxKKtD1rGxlG4LjncdabFn9gvLZad2bSysqz/qTAUStTvqJQIDAQAB
+AoGAGRzwwir7XvBOAy5tM/uV6e+Zf6anZzus1s1Y1ClbjbE6HXbnWWF/wbZGOpet
+3Zm4vD6MXc7jpTLryzTQIvVdfQbRc6+MUVeLKwZatTXtdZrhu+Jk7hx0nTPy8Jcb
+uJqFk541aEw+mMogY/xEcfbWd6IOkp+4xqjlFLBEDytgbIECQQDvH/E6nk+hgN4H
+qzzVtxxr397vWrjrIgPbJpQvBsafG7b0dA4AFjwVbFLmQcj2PprIMmPcQrooz8vp
+jy4SHEg1AkEA/v13/5M47K9vCxmb8QeD/asydfsgS5TeuNi8DoUBEmiSJwma7FXY
+fFUtxuvL7XvjwjN5B30pNEbc6Iuyt7y4MQJBAIt21su4b3sjXNueLKH85Q+phy2U
+fQtuUE9txblTu14q3N7gHRZB4ZMhFYyDy8CKrN2cPg/Fvyt0Xlp/DoCzjA0CQQDU
+y2ptGsuSmgUtWj3NM9xuwYPm+Z/F84K6+ARYiZ6PYj013sovGKUFfYAqVXVlxtIX
+qyUBnu3X9ps8ZfjLZO7BAkEAlT4R5Yl6cGhaJQYZHOde3JEMhNRcVFMO8dJDaFeo
+f9Oeos0UUothgiDktdQHxdNEwLjQf7lJJBzV+5OtwswCWA==
+-----END RSA PRIVATE KEY-----`)
+
+func startTestHTTPS(handler http.Handler, certificate []byte, privateKey []byte) *httptest.Server {
+	ts := httptest.NewUnstartedServer(handler)
+	cert, _ := tls.X509KeyPair(certificate, privateKey)
+	existingConfig := ts.TLS
+	if existingConfig != nil {
+		ts.TLS = existingConfig.Clone()
+	} else {
+		ts.TLS = new(tls.Config)
+	}
+	if ts.TLS.NextProtos == nil {
+		nextProtos := []string{"http/1.1"}
+		//if ts.EnableHTTP2 { // to be uncommented when go version is updated in build_servers
+		//	nextProtos = []string{"h2"}
+		//}
```

Comment:
I don't understand this comment; the golang version for the client is defined by yocto, no relation with the `build_servers`. Better to specify from which golang version on can we uncomment this.

Reply:
will reword that

Reply:
I chose to reword the comment. 
Alf, if you insist on removing that, I have no problem with that. I can also add:
```go
 if runtime.Version() == "go1.12" { ... }
```
I wonder if that would work? but I would like to avoid it, since it leads to good old `#ifdef __PLATFORM__ ... #endif` mess.